jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/ConfirmIamPolicyDialog.kt

```diff
@@ -0,0 +1,86 @@
+
+import com.intellij.json.JsonLanguage
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import software.amazon.awssdk.services.iam.IamClient
+import software.amazon.awssdk.services.lambda.LambdaClient
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.warn
+import software.aws.toolkits.jetbrains.services.lambda.upload.createSqsPollerPolicy
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.ui.formatAndSet
+import software.aws.toolkits.resources.message
+import java.awt.Component
+import javax.swing.JComponent
+
+class ConfirmIamPolicyDialog(
+    private val project: Project,
+    private val iamClient: IamClient,
+    private val lambdaClient: LambdaClient,
+    private val functionName: String,
+    private val queue: Queue,
+    private val parent: Component? = null
+) : DialogWrapper(project, parent, false, IdeModalityType.PROJECT), CoroutineScope by ApplicationThreadPoolScope("ConfirmIamPolicy") {
+    private val rolePolicy: String by lazy { createSqsPollerPolicy(queue.arn) }
+    private val policyName: String by lazy { "AWSLambdaSQSPollerExecutionRole-$functionName-${queue.queueName}" }
+    val view = ConfirmIamPolicyPanel(project)
+
+    init {
+        title = message("sqs.confirm.iam")
+        setOKButtonText(message("sqs.confirm.iam.create"))
+        view.policyDocument.formatAndSet(rolePolicy, JsonLanguage.INSTANCE)
+        init()
+    }
+
+    override fun createCenterPanel(): JComponent? = view.component
+
+    override fun doOKAction() {
+        if (isOKActionEnabled) {
+            setOKButtonText(message("sqs.confirm.iam.in_progress"))
+            isOKActionEnabled = false
+
+            launch {
+                try {
+                    val policyArn = createPolicy()
+                    attachPolicy(policyArn)
+                    runInEdt(ModalityState.any()) {
+                        close(OK_EXIT_CODE)
+                    }
+                } catch (e: Exception) {
+                    LOG.warn(e) { message("sqs.confirm.iam.failed") }
+                    setErrorText(e.message)
+                    setOKButtonText(message("sqs.confirm.iam.create"))
+                    isOKActionEnabled = true
+                }
+            }
+        }
+    }
+
+    private fun createPolicy(): String {
+        val policy = iamClient.createPolicy {
+            it.policyName(policyName)
+            it.policyDocument(rolePolicy)
+        }.policy()
+        return policy.arn()
+    }
+
+    private fun attachPolicy(policyArn: String) {
+        val role = lambdaClient.getFunctionConfiguration { it.functionName(functionName) }.role().substringAfterLast('/')
```

Comment:
Can you add a comment why we do substringAfterLast, like put in an example of what is returned (from the getFuncitonConfiguration call I guess I would expect back something we can use)